SofaKernel/modules/SofaBaseUtils/src/SofaBaseUtils/FromComponent.cpp

```diff
@@ -0,0 +1,136 @@
+#include <sofa/helper/system/FileSystem.h>
+using sofa::core::objectmodel::ComponentState ;
+
+#include <SofaBaseUtils/FromComponent.h>
+
+using std::string;
+
+namespace sofa::component
+{
+
+FromComponent::FromComponent() :
+    d_plugin(initData(&d_plugin, "plugin", ""))
+  ,d_import_old_name(initData(&d_import_old_name, "import", ""))
+  ,d_as_new_name(initData(&d_as_new_name, "as", ""))
+{
+    d_componentState.setValue(ComponentState::Invalid) ;
+}
+
+void FromComponent::parse ( core::objectmodel::BaseObjectDescription* arg )
+{
+    BaseObject::parse(arg) ;
+
+    const char* plugin=arg->getAttribute("plugin") ;
+    const char* object=arg->getAttribute("import") ;
+    const char* alias=arg->getAttribute("as") ;
+
+    if(plugin==nullptr)
+    {
+        msg_error() << "The mandatory 'plugin' attribute is missing.  "
+                           "The component is disabled.  "
+                           "To remove this error message you need to add a targetcomponent attribute pointing to a valid component's ClassName.";
+        return ;
+    }
+    string starget(plugin) ;
+
+    if(object==nullptr)
+    {
+        msg_error() << "The mandatory 'import' attribute is missing.  "
+                           "The component is disabled.  "
+                           "To remove this error message you need to add an alias attribute with a valid string component's ClassName.";
+        return ;
+    }
+
+    // First load the plugin if it is not yet done
+    type::vector< std::string > failed;
+    std::ostringstream errmsg;
+    const std::string name = sofa::helper::system::FileSystem::cleanPath( plugin );
+    auto& pluginManager = sofa::helper::system::PluginManager::getInstance();
+    if ( !pluginManager.pluginIsLoaded(name) )
+    {
+        if(!pluginManager.loadPlugin(name, sofa::helper::system::PluginManager::getDefaultSuffix(), true, true, &errmsg))
+        {
+            msg_error() << errmsg.str();
+        }
+    }
+
+
+    // Then import the requested names.
+    if(std::string(object)=="*")
+    {
+        std::vector<sofa::core::ObjectFactory::ClassEntry::SPtr> entries;
+        ObjectFactory::getInstance()->getEntriesFromTarget(entries, std::string(plugin));
+        for(auto& entry : entries)
+        {
+            string fullname = std::string(plugin)+"."+entry->className;
+            ObjectFactory::getInstance()->addAlias(entry->className, fullname,true);
+        }
+        if(alias != nullptr)
+        {
+            msg_error() << "using 'as' and '*' at the same time is not supposed to work.";
+            return ;
+        }
+        d_componentState.setValue(ComponentState::Valid);
+        return;
+    }
+
+    if(alias == nullptr)
+    {
+        alias = object;
+    }
+
+    std::stringstream tmp;
+    tmp << plugin << "." << object;
+    std::string fullname = tmp.str();
+    if(!ObjectFactory::getInstance()->hasCreator(fullname))
+    {
+        std::stringstream tmp;
```

Comment:
`tmp` is already declared and I recommend a more explicit name. `errorMessage` for example